OpenSim/Simulation/Model/ContactMesh.cpp

```diff
@@ -145,7 +148,7 @@ void ContactMesh::generateDecorations(bool fixed, const ModelDisplayHints& hints
         const auto& X_BF = getFrame().findTransformInBaseFrame();
         const auto& X_FP = getTransform();
         const auto X_BP = X_BF * X_FP;
-        geometry.push_back(SimTK::DecorativeMeshFile(get_filename())
+        geometry.push_back(SimTK::DecorativeMesh(*_decorativeGeometry)
```

Comment:
Ah....so is the reason that the class holds onto a `DecorativeMesh` that it is cheaper to copy construct it than to recreate it from a file for every visualization frame? Could the code look like this:

```cpp
void loadMesh() {
...
_decorativeGeometry.reset(new SimTK::DecorativeMesh(mesh).setTransform(X_BFP).setRepresentation(...)...
}

void generateDecorations(...) {
...
geometry.push_back(*_decorativeGeometry);
}
```
?

If that doesn't make sense for some reason, then it's fine to leave this as is...the unnecessary copy is bothering me though. Either way, I'm fine with how this is.